homeassistant/components/screenlogic/binary_sensor.py

```diff
@@ -1,161 +1,194 @@
+                VALUE.PH_SUPPLY_ALARM: SupportedBinarySensorValueParameters(),
+                VALUE.PROBE_FAULT_ALARM: SupportedBinarySensorValueParameters(),
+            },
+            GROUP.ALERT: {
+                VALUE.ORP_LIMIT: SupportedBinarySensorValueParameters(),
+                VALUE.PH_LIMIT: SupportedBinarySensorValueParameters(),
+                VALUE.PH_LOCKOUT: SupportedBinarySensorValueParameters(),
+            },
+            GROUP.WATER_BALANCE: {
+                VALUE.CORROSIVE: SupportedBinarySensorValueParameters(),
+                VALUE.SCALING: SupportedBinarySensorValueParameters(),
+            },
+        },
+        DEVICE.SCG: {
+            GROUP.SENSOR: {
+                VALUE.STATE: SupportedBinarySensorValueParameters(),
+            },
+        },
+    }
 )
 
+SL_DEVICE_TYPE_TO_HA_DEVICE_CLASS = {DEVICE_TYPE.ALARM: BinarySensorDeviceClass.PROBLEM}
+
 
 async def async_setup_entry(
     hass: HomeAssistant,
     config_entry: ConfigEntry,
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up entry."""
-    entities: list[ScreenLogicBinarySensorEntity] = []
-    coordinator: ScreenlogicDataUpdateCoordinator = hass.data[DOMAIN][
+    entities: list[ScreenLogicBinarySensor] = []
+    coordinator: ScreenlogicDataUpdateCoordinator = hass.data[SL_DOMAIN][
         config_entry.entry_id
     ]
-    gateway_data = coordinator.gateway_data
-    config = gateway_data[SL_DATA.KEY_CONFIG]
-
-    # Generic binary sensor
-    entities.append(
-        ScreenLogicStatusBinarySensor(coordinator, "chem_alarm", CODE.STATUS_CHANGED)
-    )
-
-    entities.extend(
-        [
-            ScreenlogicConfigBinarySensor(coordinator, cfg_sensor, CODE.STATUS_CHANGED)
-            for cfg_sensor in config
-            if cfg_sensor in SUPPORTED_CONFIG_BINARY_SENSORS
-        ]
-    )
-
-    if config["equipment_flags"] & EQUIPMENT.FLAG_INTELLICHEM:
-        chemistry = gateway_data[SL_DATA.KEY_CHEMISTRY]
-        # IntelliChem alarm sensors
-        entities.extend(
-            [
-                ScreenlogicChemistryAlarmBinarySensor(
-                    coordinator, chem_alarm, CODE.CHEMISTRY_CHANGED
+    gateway = coordinator.gateway
+    data_path: ScreenLogicDataPath
+    value_params: SupportedBinarySensorValueParameters
+    for data_path, value_params in iterate_expand_group_wildcard(
+        gateway, SUPPORTED_DATA
+    ):
+        entity_key = generate_unique_id(*data_path)
+
+        device = data_path[0]
+
+        if not (DEVICE_INCLUSION_RULES.get(device) or value_params.included).test(
+            gateway, data_path
+        ):
+            cleanup_excluded_entity(coordinator, DOMAIN, entity_key)
+            continue
+
+        try:
+            value_data = gateway.get_data(*data_path, strict=True)
+        except KeyError:
+            _LOGGER.debug("Failed to find %s", data_path)
+            continue
+
+        entity_kwargs = {
```

Comment:
```suggestion
        description_kwargs = {
```

or `entity_description`